packages/vx-xychart/src/components/axis/BaseAxis.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useMemo, useContext } from 'react';
+import { AxisScale } from '@vx/axis';
+import { AxisProps as VxAxisProps } from '@vx/axis/lib/axis/Axis';
+import DataContext from '../../context/DataContext';
+
+export type BaseAxisProps<Scale extends AxisScale> = Omit<
+  VxAxisProps<Scale>,
+  'scale' | 'orientation'
+> & {
+  /** Required axis orientation. */
+  orientation: NonNullable<VxAxisProps<Scale>['orientation']>;
+} & {
+  /** Rendered component which is passed VxAxisProps by BaseAxis after processing. */
+  AxisComponent: React.FC<VxAxisProps<Scale>>;
+};
+
+/**
+ * Component which handles all xychart-specific logic for axes,
+ * and passes processed props to a specified Axis / AnimatedAxis component.
+ */
+export default function BaseAxis<Scale extends AxisScale>({
+  AxisComponent,
+  ...props
+}: BaseAxisProps<Scale>) {
+  const { theme, xScale, yScale, margin, width, height } = useContext(DataContext);
+  const { orientation } = props;
+
+  const axisStyles = useMemo(
+    () =>
+      orientation === 'left' || orientation === 'right'
+        ? theme?.axisStyles?.y?.[orientation]
+        : theme?.axisStyles?.x?.[orientation],
+    [theme, orientation],
+  );
+
+  const tickLabelProps = useMemo(
+    () =>
+      props.tickLabelProps || // use props if passed
+      (axisStyles // else construct from theme if possible
+        ? () =>
+            orientation === 'left' || orientation === 'right'
+              ? // by default, wrap vertical-axis tick labels within the allotted margin space
+                // this does not currently account for axis label
+                { ...axisStyles.tickLabel, width: margin?.[orientation] }
```

Comment:
this logic could prob be made smarter in the future, and extended to `labelProps`, but didn't implement for now.